### DISCUSSION TITLE
[Feature Request] Sticky Scrolling ala iTerm2

### DISCUSSION THREAD

**Discussion by Stanzilla** [ghostty-org/ghostty#3603] [Feature Requests, Ideas]
iTerm has this neat feature where the last command you used is displayed in a small bar below the tab name so if the output is long, you don't have to scroll back up to see what you ran. VSCode also has this and calls it sticky scroll, I don't really know what the name is but I would love to see it in Ghostty.

![iTerm-2024-12-27-23 42 37](https://github.com/user-attachments/assets/c4485484-6ef6-4020-939c-b06f03f5d887)

**Comment by Integralist** [comment]
> iTerm has this neat feature where the last command you used is displayed in a small bar below the tab name so if the output is long, you don't have to scroll back up to see what you ran. VSCode also has this and calls it sticky scroll, I don't really know what the name is but I would love to see it in Ghostty.
> 
> ![iTerm-2024-12-27-23 42 37](https://github.com/user-attachments/assets/c4485484-6ef6-4020-939c-b06f03f5d887)
> 
> 
> 

Warp also has this feature and I agree is quite nice 👍

**Reply by JakeWharton** [reply]
In Warp you can click that pinned bar to scroll to the top of the output (basically where the pinned section would natively reside with the output starting below it).

I miss this feature every day.

**Comment by mitchellh** [comment]
I like this idea a lot. I want to think about the design (technical) a bit more before converting to an issue, but just giving you some feedback I like it.

**Reply by andrew-ares** [reply]
Hi, just checking if this feature is still on your radar.
